packages/cli/src/commands/init/template.ts

```diff
@@ -68,10 +70,26 @@ export async function copyTemplate(
     templateName,
     templateDir,
   );
+  const projectDirectory = process.cwd();
+
+  const projectDirectoryFiles = await getDirectoryFilesRecursive(
+    projectDirectory,
+  );
+  const templatePathFiles = await getDirectoryFilesRecursive(templatePath);
+
+  const conflictingFiles: string[] = [];
+  for (const templatePathFile of templatePathFiles) {
+    if (projectDirectoryFiles.includes(templatePathFile)) {
+      conflictingFiles.push(path.join(projectDirectory, templatePathFile));
+    }
+  }
+  if (conflictingFiles.length > 0) {
+    throw new ConflictingFilesError(projectDirectory, conflictingFiles);
+  }
```

Comment:
I have a conceptual suggestion.

How about we move all this code into a function `checkDirectoryForConflictingFiles` (name is just a recommendation) and call it from an `else` block of this conditional: https://github.com/react-native-community/cli/pull/1446/files#diff-5b5591728aa944b7a0f0a9a01b281bdf3e9f51bded73219124ace0edad8afd7eR48

Basically, I want to minimise the number of files affected. If directory exists - validate for conflicts - otherwise - create it. Everything stays in a single conditional.